android-vjestina-homework-movie-app-data-mock/app/src/main/java/agency/five/codebase/android/movieapp/ui/main/MainScreen.kt

```diff
@@ -0,0 +1,199 @@
+                    HomeRoute(
+                        onNavigateToMovieDetails = {
+                            navController.navigate(
+                                MovieDetailsDestination.createNavigationRoute(it.id)
+                            )
+                        }
+                    )
+                }
+
+                composable(NavigationItem.FavoritesDestination.route) {
+                    FavoritesRoute(
+                        onNavigateToMovieDetails = {
+                            navController.navigate(
+                                MovieDetailsDestination.createNavigationRoute(it.id)
+                            )
+                        }
+                    )
+                }
+
+                composable(
+                    route = MovieDetailsDestination.route,
+                    arguments = listOf(navArgument(MOVIE_ID_KEY) { type = NavType.IntType }),
+                ) {
+                    MovieDetailsRoute()
+                }
+            }
+        }
+    }
+}
+
+@Composable
+private fun TopBar(
+    navigationIcon: @Composable (() -> Unit)? = null,
+) {
+    TopAppBar(
+        backgroundColor = MaterialTheme.colorScheme.primary,
+        contentColor = MaterialTheme.colorScheme.onPrimary,
+        modifier = Modifier
+            .fillMaxHeight(0.07f)
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxSize(1f),
+            contentAlignment = Alignment.Center,
+        ) {
+            Image(
+                painter = painterResource(id = R.drawable.tmdb_logo),
+                contentDescription = null,
+            )
+            if (navigationIcon != null) {
+                navigationIcon()
+            }
+        }
+    }
+}
+
+@Composable
+private fun BackIcon(
+    onBackClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    Box(
+        modifier = modifier
+            .fillMaxSize(1f),
+        contentAlignment = Alignment.CenterStart
+    ) {
+        Icon(
+            painter = painterResource(id = R.drawable.back_icon),
+            contentDescription = null,
+            modifier = Modifier
+                .padding(8.dp)
+                .clickable(onClick = onBackClick)
+        )
+    }
+}
+
+@Composable
+private fun BottomNavigationBar(
+    destinations: List<NavigationItem>,
+    onNavigateToDestination: (NavigationItem) -> Unit,
+    currentDestination: NavDestination?,
+) {
+    BottomNavigation(
+        backgroundColor = MaterialTheme.colorScheme.background
+    ) {
+        destinations.forEach { destination ->
+            BottomNavigationItem(
+                selected = currentDestination?.route == destination.route,
+                icon = {
+                    Icon(
+                        painter = painterResource(
+                            id = if (currentDestination?.route == destination.route)
+                                destination.selectedIconId
```

Comment:
Duplication of `currentDestination?.route == destination.route`.
Extract it to a variable (DRY)

Reply:
How to extract it when it's inside destinations.forEach which iterates through NavigationItem list?